meson.build

```diff
@@ -0,0 +1,17 @@
+#              Copyright Catch2 Authors
+# Distributed under the Boost Software License, Version 1.0.
+#   (See accompanying file LICENSE_1_0.txt or copy at
+#        https://www.boost.org/LICENSE_1_0.txt)
+
+# SPDX-License-Identifier: BSL-1.0
+
+project(
+  'catch2',
+  'cpp',
+  version : '3.1.0',
```

Comment:
This should either slurp the version from the source files, or the update script in `tools/scripts/releaseCommon.py` needs to be updated to update this file as well.

Reply:
Related, does Meson support the idea of "in-development" versions? I generally dislike the fact that we use the 3.1.0 version number for what is the development version of next release, but most tools don't support development versions too well.

Reply:
I think that there are two alternatives here:

1. Call a script that uses the code in `releaseCommon.py` to read the version number out of `catch_version.cpp`. (This may require a more recent version of Meson than is in Debian 11, which complicates things.)
2. Teach `releaseCommon.py` to also put the version number in `meson.build`. This looks like it would be straightforward and Meson would be no worse that CMake.

Regarding in-development versions:

It looks like if we were to require a sufficiently-new version of Meson (newer than is in Debian 11) then it would be possible to use something like:

```meson
project(
  'catch2',
  'cpp',
  version : run_command('git', 'describe', check: true).stdout().strip(),
  license: 'BSL-1.0',
  meson_version: '>=0.49.0'
)
```
so that the version number ends up being generated automatically by Git. The problem with this is that the release scripts would need to patch this with the actual version when producing the tarball and ZIP file releases.

See also: https://stackoverflow.com/questions/59201214/can-the-meson-project-version-be-assigned-dynamically

Reply:
I believe that I've implemented option 2.

Reply:
Yeah, given that's what we already do for other places, adding it to `releaseCommon` was the right choice.